protocol/mqtt_paho/v2/protocol.go

```diff
@@ -41,96 +41,101 @@ var (
 	_ protocol.Closer   = (*Protocol)(nil)
 )
 
-func New(ctx context.Context, clientConfig *paho.ClientConfig, connConfig *paho.Connect, SenderTopic string,
-	ReceiverTopics []string, qos byte, retained bool,
-) (*Protocol, error) {
-	client := paho.NewClient(*clientConfig)
-	ca, err := client.Connect(ctx, connConfig)
-	if err != nil {
+func New(config *paho.ClientConfig, opts ...Option) (*Protocol, error) {
```

Comment:
questions:

- should we keep `ctx` as first arg?
- can we supply defaults to `config` so it can also become an option? I see it's mostly used to set a client `id` and `conn` which can be defaulted IMHO

Reply:
- We can keep the `ctx` as the first args. So I will init the client on the `New()`. 
- Of course, we can set the `id` by default. but how can we set a default `conn` from the `config`?  If we don't specify the address of the conn, it seems there is no way to establish a connection. and throw the error message: `client connection is nil`